src/library_pthread.js

```diff
@@ -377,6 +377,12 @@ var LibraryPThread = {
     pthread.worker.postMessage({ cmd: 'cancel' });
   },
 
+  _global_locale_ptr: 0,
+  __emscripten_setup_pthread_data__deps: ['_global_locale_ptr'],
+  __emscripten_setup_pthread_data: function(global_locale_ptr){
+    __global_locale_ptr = global_locale_ptr|0;
+  },
```

Comment:
What does the globalness here refer to? Does this need to be global across multiple threads? Because variables in JS are limited to being thread local, so each pthread will have its own copy of `_global_locale_ptr`. If this needs to be global across threads, the variable should live in `library_pthread.c`.